plugins/inputs/mongodb/mongodb_server.go

```diff
@@ -147,11 +147,48 @@ func (s *Server) gatherData(acc telegraf.Accumulator, gatherDbStats bool) error
 		}
 	}
 
+	result_col_stats := &ColStats{}
+	if gatherColStats == true {
+		names := []string{}
+		names, err = s.Session.DatabaseNames()
+		if err != nil {
+			log.Println("E! Error getting database names (" + err.Error() + ")")
+		}
+		for _, db_name := range names {
+			if stringInSlice(db_name, colStatsDbs) || len(colStatsDbs) == 0 {
+				var colls []string
+				colls, err = s.Session.DB(db_name).CollectionNames()
+				if err != nil {
+					log.Println("E! Error getting collection names (" + err.Error() + ")")
+				}
+				for _, col_name := range colls {
+					col_stat_line := &ColStatsData{}
+					err = s.Session.DB(db_name).Run(bson.D{
+						{
+							Name:  "collStats",
+							Value: col_name,
+						},
+					}, col_stat_line)
+					if err != nil {
+						log.Println("E! Error getting col stats from " + col_name + "(" + err.Error() + ")")
```

Comment:
Should fmt.Errorf be used here to return an error and prevent further execution?

Reply:
I think maybe a `continue` so it continues adding other (likely valid) colls

Reply:
17465b0aaf2489eac8ec1a5a9b72ed32f7d2ebf1

Reply:
Yeah, also we need to do something on 155.  Probably should move this into a function.